app.rb

```diff
@@ -79,12 +79,18 @@ def config
         id_token = token_response[:id_token]
         userinfo_response = userinfo(id_token)
 
-        session[:login_msg] = 'ok'
-        session[:logout_uri] = logout_uri(token_response[:id_token])
-        session[:userinfo] = userinfo_response
-        session[:email] = session[:userinfo][:email]
+        if session.delete(:step_up_enabled)
+          aal = session.delete(:step_up_aal)
```

Comment:
Could we just have the one session value?

```suggestion
        if (aal = session.delete(:step_up_aal))
```

Reply:
Since aal could be nil, I'm going to leave it as is.